conans/cli/commands/list.py

```diff
@@ -1,67 +1,83 @@
-            cli_out_write("{}There are no matching recipes".format(" " * indentation))
 
+def list_recipes_cli_formatter(results):
+    for remote_results in results:
+        _print_common_list_cli_output(remote_results, "recipes")
         current_recipe = None
         for recipe in remote_results["results"]:
             if recipe["name"] != current_recipe:
                 current_recipe = recipe["name"]
-                cli_out_write("{}{}".format(" " * indentation, current_recipe), recipe_color)
+                cli_out_write(current_recipe, fg=recipe_color, indentation=2)
 
             reference = recipe["id"]
-            cli_out_write("{}{}".format(" " * (indentation * 2), reference), reference_color)
+            cli_out_write(reference, fg=reference_color, indentation=4)
 
 
-def list_recipe_revisions_cli_formatter(results):
+def _list_revisions_cli_formatter(results, ref_type):
     for remote_results in results:
-        if remote_results.get("error"):
-            # TODO: Handle errors
-            return
+        _print_common_list_cli_output(remote_results, ref_type)
+        reference = remote_results["reference"]
+        for revisions in remote_results["results"]:
+            rev = revisions["revision"]
+            date = iso8601_to_str(revisions["time"])
+            cli_out_write(f"{reference}#{rev} ({date})", fg=recipe_color, indentation=2)
 
-        if not remote_results.get("remote"):
-            cli_out_write("Local Cache:", remote_color)
-        else:
-            cli_out_write("{}:".format(remote_results["remote"]), remote_color)
 
-        if not remote_results.get("results"):
-            cli_out_write("{}There are no matching recipes".format(" " * indentation))
+def list_recipe_revisions_cli_formatter(results):
+    _list_revisions_cli_formatter(results, "recipes")
 
-        reference = remote_results["reference"]
-        for revisions in remote_results["results"]:
-            cli_out_write(
-                "{}{}#{} ({})".format(
-                    " " * indentation,
-                    reference,
-                    revisions["revision"],
-                    iso8601_to_str(revisions["time"])
-                ),
-                recipe_color
-            )
+
+def list_package_revisions_cli_formatter(results):
+    _list_revisions_cli_formatter(results, "packages")
 
 
+def list_package_ids_cli_formatter(results):
+    # Artifactory uses field 'requires', conan_center 'full_requires'
+    requires_fields = ("requires", "full_requires")
+    general_fields = ("options", "settings")
+
+    for remote_results in results:
+        _print_common_list_cli_output(remote_results, "references")
+        reference = remote_results["reference"]
+        for pkg_id, props in remote_results["results"].items():
+            cli_out_write(repr(PackageReference(reference, pkg_id)), fg=reference_color, indentation=2)
+            for prop_name, values in props.items():
+                if prop_name in requires_fields:
```

Comment:
I think we should check here if `values` is empty or not so that it does not print the title like `requires:` if there are no requires.

Reply:
Sorry for reopening this, but I think it would be cleaner if you check `values` only once:

```python
for prop_name, values in props.items():
    if not values:
        continue

    if prop_name in requires_fields:
        blablabla
    elif prop_name in general_fields:
        blablabla
```